### PR TITLE
[14.0][FIX] delivery_price_product_domain: fix shipping without rule

### PR DESCRIPTION
**Pull request by dessanhemrayev**

### CODE REVIEW

File: delivery_price_product_domain/models/delivery_carrier.py [hunk 1]
```diff
@@ -52,11 +52,12 @@ def recompute_price_available(self, apply_product, price_dict, untaxed_in_dict):
 
     def _get_price_from_picking(self, total, weight, volume, quantity):
         "Find price rule matching SO base on apply_product_domain"
-        if self.free_over and total >= self.amount:
-            return 0
+        if self.free_over and total >= self.amount or not self.price_rule_ids:
```
Comment (rousseldenis) [inline delivery_price_product_domain/models/delivery_carrier.py]:
```suggestion
        if (self.free_over and total >= self.amount) or not self.price_rule_ids:
```

**Review by HekkiMelody — APPROVED** [review]
Code review

**Review by francesco-ooops — APPROVED** [review]
Functional ok

@rousseldenis what do you say?

**Comment by francesco-ooops** [comment]
@pedrobaeza can we merge?

**Comment by pedrobaeza** [comment]
@rousseldenis should confirm

**Comment by francesco-ooops** [comment]
@rousseldenis does it look good for you? thanks!

File: delivery_price_product_domain/models/delivery_carrier.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def _get_price_from_picking(self, total, weight, volume, quantity):
         price_dict = self._get_price_dict(total, weight, volume, quantity)
         untaxed_in_dict = "untaxed_price" in price_dict
         test = False
-        rule_line = self.price_rule_ids[0]
+        rule_line = False
```
Comment (rousseldenis) [inline delivery_price_product_domain/models/delivery_carrier.py]:
```suggestion
        rule_line = self.price_rule_ids.browse()
```

is better.

**Review by rousseldenis — APPROVED** [review]

**Comment by francesco-ooops** [comment]
@chienandalu as author of https://github.com/OCA/delivery-carrier/tree/14.0/delivery_cttexpress , could you help us understand whether the error in tests is temporary or needs fixing? thanks!

**Comment by rousseldenis** [comment]
@francesco-ooops https://github.com/OCA/odoo-community.org/pull/158#issuecomment-1999174253

**Comment by francesco-ooops** [comment]
> @francesco-ooops [OCA/odoo-community.org#158 (comment)](https://github.com/OCA/odoo-community.org/pull/158#issuecomment-1999174253)

Thanks, very interesting topic indeed!

**Pull request merged**